cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/provisioner/TetheringConf.java

```diff
@@ -53,6 +56,20 @@ private static String getString(Map<String, String> properties, String key) {
     return val;
   }
 
+  private static void validateConfig(String peer, String namespace, TetheringStore tetheringStore) {
+    PeerInfo peerInfo;
+    try {
+      peerInfo = tetheringStore.getPeer(peer);
+    } catch (Exception e) {
+      throw new IllegalArgumentException(String.format("%s is not a tethered peer", peer));
+    }
+    if (peerInfo.getMetadata().getNamespaceAllocations().stream()
+      .noneMatch(n -> n.getNamespace().equals(namespace))) {
+      throw new IllegalArgumentException(String.format("Namespace %s is not provided by tethered peer %s",
```

Comment:
is this checking the same thing that the TetheringAgentService checks, except earlier on the CDF side?

Reply:
Yes. Compute profile creation will fail on CDF if this check fails.